### PR TITLE
Unit tests fo JDropDownMenu class

### PR DESCRIPTION
**Pull request by Auriga2**
### Prerequisites

- [x] Reviewed the [checklist](CHECKLIST.md)

- [x] Reviewed feedback from the "Sonar Cloud" bot. Note that you have to wait
    for the "CI / Unit Tests") to complete first.

### Description of the Change
Unit tests fo JDropDownMenu class

### Verification Process
All tests pass

### Applicable Issues
#413

### CODE REVIEW

**Review by antares1470 — APPROVED** [review]

File: CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/gui/JDropDownMenuNGTest.java [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+
+    /**
+     * Test the constructor works with an Icon
+     */
+    @Test
+    public void testConstructor_WithIcon() {
+        final Icon icon = new ImageIcon(COOKIE_ICON_PATH);
+        JDropDownMenu instance = new JDropDownMenu(icon, items);
+        Icon result = instance.getIcon();
+        assertNotNull(result);
```
Comment (aldebaran30701) [inline CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/gui/JDropDownMenuNGTest.java:110]: Would you be able to compare that the icon is the same?
ie. `assertEquals(result, icon);`

File: CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/gui/JDropDownMenuNGTest.java [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+    }
+
+    /**
+     * Test of getIcon method, of class JDropDownMenu.
+     */
+    @Test
+    public void testGetIcon() {
+        System.out.println("getIcon");
+        JDropDownMenu instance = new JDropDownMenu("Text", items);
+        Icon result = instance.getIcon();
```
Comment (aldebaran30701) [inline CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/gui/JDropDownMenuNGTest.java:145]: Also checking the Icon is the proper icon here

File: CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/gui/JDropDownMenuNGTest.java [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+
+    /**
+     * Test of getItems method, of class JDropDownMenu.
+     */
+    @Test
+    public void testGetItems() {
+        System.out.println("getItems");
+        JDropDownMenu instance = new JDropDownMenu("Text", items);
+        Set result = instance.getItems();
+        assertEqualsNoOrder(result.toArray(), items.toArray());
```
Comment (aldebaran30701) [inline CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/gui/JDropDownMenuNGTest.java:181]: Nice! I'll be using this assert more often, was looking for something like this.

**Review by aldebaran30701 — COMMENTED** [review]
Just a minor comment

File: CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/gui/JDropDownMenuNGTest.java (hunk 1 above)
Comment (Auriga2) [reply CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/gui/JDropDownMenuNGTest.java:110]: No they were `CompoundIcon`s and I couldn't compare them to be equal.

File: CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/gui/JDropDownMenuNGTest.java (hunk 2 above)
Comment (Auriga2) [reply CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/gui/JDropDownMenuNGTest.java:145]: Here they appeared to be equal but none of the assert methods asserted. So I ended up checking the existence.

File: CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/gui/JDropDownMenuNGTest.java (hunk 1 above)
Comment (antares1470) [reply CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/gui/JDropDownMenuNGTest.java:110]: You could potentially check if the Icon added is contained in the CompoundIcon (casting `result` as `CompoundIcon`)?

File: CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/gui/JDropDownMenuNGTest.java (hunk 2 above)
Comment (Auriga2) [reply CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/gui/JDropDownMenuNGTest.java:145]: It works in the `testSetIcon` for whatever the Icon we set. So I guess this is fine, we can even get rid of this test...

File: CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/gui/JDropDownMenuNGTest.java (hunk 1 above)
Comment (Auriga2) [reply CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/gui/JDropDownMenuNGTest.java]: Yes it worked. Thanks Josh.

File: CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/gui/JDropDownMenuNGTest.java (hunk 2 above)
Comment (Auriga2) [reply CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/gui/JDropDownMenuNGTest.java:153]: This method is actually covered by `testSetIcon` and `testConstructor_WithIcon` in two different ways. Should I copy paste one of them here or just remove the test?

**Review by aldebaran30701 — APPROVED** [review]

**Review by antares1470 — APPROVED** [review]

**Pull request merged**